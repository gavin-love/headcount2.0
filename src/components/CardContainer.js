import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import  '../styles/CardContainer.css'

const CardContainer = ({districts, findAverages, selectedDistricts}) => {
  const allDistricts = districts.map((district, index) => {
    // console.log(selectedDistricts)
    return <Card location={district.Location} stats={district.stats} key={index} selected={selectedDistricts} findAverages={findAverages} />
  })
  return (
    <div className="card-container">
    {allDistricts}
    </div>
  )
}

export default CardContainer;