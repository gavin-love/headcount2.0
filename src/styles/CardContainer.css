.card-container {
  background-color: #BACACB;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding-top: 10px;
}