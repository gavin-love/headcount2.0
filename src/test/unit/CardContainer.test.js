import React from "react";
import CardContainer from "../../components/CardContainer.js";
import { shallow } from "enzyme";

describe("CardContainer", () => {
  let wrapper;
  let mockData;
  let mockFunction;

  beforeEach(() => {
    mockData = [
      { Location: "colorado", stats: { 2004: 0.24, 2005: 0.63, 2006: 0.11 } },
      { Location: "academy 20", stats: { 2012: 0.24, 2013: 0.63, 2014: 0.11 } }
    ];
    mockFunction = jest.fn();
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should", () => {
    wrapper = shallow(
      <CardContainer districts={mockData} findAverages={mockFunction} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
