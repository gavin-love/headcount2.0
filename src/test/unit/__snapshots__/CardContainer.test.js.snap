// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `undefined`;

exports[`Card should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districts={
        Array [
            Object {
              "Location": "colorado",
              "stats": Object {
                "2004": 0.24,
                "2005": 0.63,
                "2006": 0.11,
              },
            },
            Object {
              "Location": "academy 20",
              "stats": Object {
                "2012": 0.24,
                "2013": 0.63,
                "2014": 0.11,
              },
            },
          ]
    }
    findAverages={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          findAverages={[Function]}
          location="colorado"
          stats={
                    Object {
                              "2004": 0.24,
                              "2005": 0.63,
                              "2006": 0.11,
                            }
          }
/>,
        <Card
          findAverages={[Function]}
          location="academy 20"
          stats={
                    Object {
                              "2012": 0.24,
                              "2013": 0.63,
                              "2014": 0.11,
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "findAverages": [Function],
          "location": "colorado",
          "stats": Object {
            "2004": 0.24,
            "2005": 0.63,
            "2006": 0.11,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "findAverages": [Function],
          "location": "academy 20",
          "stats": Object {
            "2012": 0.24,
            "2013": 0.63,
            "2014": 0.11,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            findAverages={[Function]}
            location="colorado"
            stats={
                        Object {
                                    "2004": 0.24,
                                    "2005": 0.63,
                                    "2006": 0.11,
                                  }
            }
/>,
          <Card
            findAverages={[Function]}
            location="academy 20"
            stats={
                        Object {
                                    "2012": 0.24,
                                    "2013": 0.63,
                                    "2014": 0.11,
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "findAverages": [Function],
            "location": "colorado",
            "stats": Object {
              "2004": 0.24,
              "2005": 0.63,
              "2006": 0.11,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "findAverages": [Function],
            "location": "academy 20",
            "stats": Object {
              "2012": 0.24,
              "2013": 0.63,
              "2014": 0.11,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer matches the snapshot 1`] = `undefined`;

exports[`CardContainer should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districts={
        Array [
            Object {
              "Location": "colorado",
              "stats": Object {
                "2004": 0.24,
                "2005": 0.63,
                "2006": 0.11,
              },
            },
            Object {
              "Location": "academy 20",
              "stats": Object {
                "2012": 0.24,
                "2013": 0.63,
                "2014": 0.11,
              },
            },
          ]
    }
    findAverages={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          findAverages={[Function]}
          location="colorado"
          stats={
                    Object {
                              "2004": 0.24,
                              "2005": 0.63,
                              "2006": 0.11,
                            }
          }
/>,
        <Card
          findAverages={[Function]}
          location="academy 20"
          stats={
                    Object {
                              "2012": 0.24,
                              "2013": 0.63,
                              "2014": 0.11,
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "findAverages": [Function],
          "location": "colorado",
          "stats": Object {
            "2004": 0.24,
            "2005": 0.63,
            "2006": 0.11,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "findAverages": [Function],
          "location": "academy 20",
          "stats": Object {
            "2012": 0.24,
            "2013": 0.63,
            "2014": 0.11,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            findAverages={[Function]}
            location="colorado"
            stats={
                        Object {
                                    "2004": 0.24,
                                    "2005": 0.63,
                                    "2006": 0.11,
                                  }
            }
/>,
          <Card
            findAverages={[Function]}
            location="academy 20"
            stats={
                        Object {
                                    "2012": 0.24,
                                    "2013": 0.63,
                                    "2014": 0.11,
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "findAverages": [Function],
            "location": "colorado",
            "stats": Object {
              "2004": 0.24,
              "2005": 0.63,
              "2006": 0.11,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "findAverages": [Function],
            "location": "academy 20",
            "stats": Object {
              "2012": 0.24,
              "2013": 0.63,
              "2014": 0.11,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
